mirascope/core/base/_utils/_convert_messages_to_message_params.py

```diff
@@ -0,0 +1,92 @@
+from __future__ import annotations
+
+from collections.abc import Sequence
+from typing import TYPE_CHECKING
+
+from typing_extensions import TypeIs
+
+from .. import AudioPart, BaseMessageParam, CacheControlPart, ImagePart, TextPart
+from .._utils import get_image_type
+
+if TYPE_CHECKING:
+    from PIL import Image
+
+    has_pil_module: bool
+else:
+    try:
+        from PIL import Image  # pyright: ignore [reportAssignmentType]
+
+        has_pil_module = True
+    except ImportError:  # pragma: no cover
+        has_pil_module = False
+
+        class Image:
+            class Image:
+                def tobytes(self) -> bytes: ...
+
+
+def _convert_message_sequence_part_to_content_part(
+    message_sequence_part: str | Image.Image | TextPart | ImagePart | AudioPart,
+) -> TextPart | ImagePart | AudioPart | CacheControlPart:
+    if isinstance(message_sequence_part, str):
+        return TextPart(text=message_sequence_part, type="text")
+    elif isinstance(
+        message_sequence_part, TextPart | ImagePart | AudioPart | CacheControlPart
+    ):
+        return message_sequence_part
+    elif has_pil_module and isinstance(message_sequence_part, Image.Image):
+        image = message_sequence_part.tobytes()
+        return ImagePart(
+            type="image",
+            media_type=f"image/{get_image_type(image)}",
+            image=image,
+            detail=None,
+        )
+    else:
+        raise ValueError(f"Invalid message sequence type: {message_sequence_part}")
+
+
+def get_content_from_message(
```

Comment:
i don't understand why this is a separate method from `_convert_message_sequence_to_content`

shouldn't we just have a single function named something like `convert_message_content_to_message_param_content`?